### PR TITLE
e2e: Update Cypress

### PR DESCRIPTION
**Pull request by Phrynobatrachus**
#### Summary
Updates Cypress and RTL dependencies. Also contains a couple changes for tests that continue to be flaky. 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-41527

### CODE REVIEW

File: tests-e2e/cypress.config.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+const {defineConfig} = require('cypress');
+
+const cypressPlugins = require('./cypress/plugins/index');
+
+module.exports = defineConfig({
```
Comment (trilopin) [inline tests-e2e/cypress.config.js:7]: has any config changed or just taken the new filename convention?

File: tests-e2e/package.json [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
     "@cypress/skip-test": "^2.6.1",
-    "@testing-library/cypress": "7.0.6",
+    "@testing-library/cypress": "^8.0.3",
     "authenticator": "^1.1.5",
     "axios": "^0.21.1",
     "axios-retry": "^3.1.9",
-    "cypress": "7.7.0",
+    "cypress": "^10.3.0",
     "cypress-file-upload": "5.0.8",
     "cypress-multi-reporters": "^1.5.0",
```
Comment (trilopin) [inline tests-e2e/package.json:10]: I'm surprised that cypress plugins are kept in old versions.

**Review by trilopin — COMMENTED** [review]
🎉 This is great @Phrynobatrachus!

I understood that all specs and the folder would need to be renamed to follow the new convention. Why is this not needed in your PR?

File: tests-e2e/cypress.config.js (hunk 1 above)
Comment (Phrynobatrachus) [reply tests-e2e/cypress.config.js:7]: Actually that's not quite right, some of the config options moved to an `e2e` property. No functional changes though.

**Comment by Phrynobatrachus** [comment]
> I understood that all specs and the folder would need to be renamed to follow the new convention. Why is this not needed in your PR?

My read of the [migration guide](https://docs.cypress.io/guides/references/migration-guide#testFiles-specPattern) is that while there's new defaults, there's no hard requirement to follow a particular structure. All that matters is that the config `specPattern` globs the tests you want.

File: tests-e2e/package.json (hunk 2 above)
Comment (Phrynobatrachus) [reply tests-e2e/package.json:10]: All are on the latest major versions. I will update them though, as I did mean to remove `cypress-file-upload`.

**Review by trilopin — APPROVED** [review]
Thanks @Phrynobatrachus for taking the time for this.

it's really cool!

**Review by lieut-data — APPROVED** [review]

**Pull request merged**